[Scala 3] Document outline not working for union types

**Describe the bug**

It seems that for example in `t.body.syntax` `.syntax` is using wrong dialect and throwing an exception in some case like union types

```
java.util.concurrent.CompletionException: java.lang.UnsupportedOperationException: Scala212 doesn't support and types
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1(CancelTokens.scala:40)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$future$1$adapted(CancelTokens.scala:38)
	at scala.concurrent.impl.CallbackRunnable.run(Promise.scala:64)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.lang.UnsupportedOperationException: Scala212 doesn't support and types
	at scala.meta.internal.prettyprinters.TreeSyntax$SyntaxInstances.$anonfun$syntaxTree$1(TreeSyntax.scala:692)
	at scala.meta.prettyprinters.Syntax$$anon$1.apply(Syntax.scala:9)
	at scala.meta.internal.prettyprinters.TreeSyntax$.$anonfun$apply$1(TreeSyntax.scala:1281)
	at scala.meta.prettyprinters.Syntax$$anon$1.apply(Syntax.scala:9)
	at scala.meta.prettyprinters.Api$XtensionSyntax.syntax(Api.scala:11)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:243)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.continue$1(DocumentSymbolProvider.scala:101)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:122)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.continue$1(DocumentSymbolProvider.scala:101)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.newOwner$1(DocumentSymbolProvider.scala:105)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:177)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1(SimpleTraverser.scala:5)
	at scala.meta.transversers.SimpleTraverser.$anonfun$apply$1$adapted(SimpleTraverser.scala:5)
	at scala.collection.immutable.List.foreach(List.scala:431)
	at scala.meta.transversers.SimpleTraverser.apply(SimpleTraverser.scala:5)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.continue$1(DocumentSymbolProvider.scala:101)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.apply(DocumentSymbolProvider.scala:122)
	at scala.meta.internal.parsing.DocumentSymbolProvider$SymbolTraverser.symbols(DocumentSymbolProvider.scala:54)
	at scala.meta.internal.parsing.DocumentSymbolProvider.$anonfun$documentSymbols$1(DocumentSymbolProvider.scala:33)
	at scala.Option.map(Option.scala:230)
	at scala.meta.internal.parsing.DocumentSymbolProvider.documentSymbols(DocumentSymbolProvider.scala:32)
	at scala.meta.internal.metals.MetalsLanguageServer.$anonfun$documentSymbol$1(MetalsLanguageServer.scala:1306)
	at scala.meta.internal.metals.CancelTokens$.$anonfun$apply$2(CancelTokens.scala:26)
	at scala.concurrent.Future$.$anonfun$apply$1(Future.scala:659)
	at scala.util.Success.$anonfun$map$1(Try.scala:255)
	at scala.util.Success.map(Try.scala:213)
	at scala.concurrent.Future.$anonfun$map$1(Future.scala:292)
	at scala.concurrent.impl.Promise.liftedTree1$1(Promise.scala:33)
	at scala.concurrent.impl.Promise.$anonfun$transform$1(Promise.scala:33)
	... 4 more
```

**To Reproduce**

Steps to reproduce the behavior, try to checkout the project and open:

https://github.com/scala/scala3-example-project/blob/master/src/main/scala/UnionTypes.scala

**Expected behavior**

Document outline is shown.

**Search terms**

document outline scala3

--- a/metals/src/main/scala/scala/meta/internal/parsing/DocumentSymbolProvider.scala
+++ b/metals/src/main/scala/scala/meta/internal/parsing/DocumentSymbolProvider.scala
@@ -5,6 +5,7 @@ import java.util.concurrent.atomic.AtomicBoolean
 
 import scala.meta._
 import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.trees.Origin
 import scala.meta.transversers.SimpleTraverser
 
 import org.eclipse.lsp4j.DocumentSymbol
@@ -30,7 +31,13 @@ class DocumentSymbolProvider(trees: Trees) {
   ): Either[util.List[DocumentSymbol], util.List[l.SymbolInformation]] = {
     val result = for {
       tree <- trees.get(path)
-    } yield new SymbolTraverser().symbols(tree).asScala
+    } yield {
+      implicit val dialect = tree.origin match {
+        case Origin.Parsed(_, dialect, _) => dialect
+        case Origin.None => dialects.Scala213
+      }
+      new SymbolTraverser().symbols(tree).asScala
+    }
     val symbols = result.getOrElse(Nil)
 
     if (supportsHierarchicalDocumentSymbols.get()) {
@@ -41,7 +48,8 @@ class DocumentSymbolProvider(trees: Trees) {
     }
   }
 
-  private class SymbolTraverser() extends SimpleTraverser {
+  private class SymbolTraverser(implicit dialect: Dialect)
+      extends SimpleTraverser {
     var owner: DocumentSymbol = new DocumentSymbol(
       "root",
       SymbolKind.Namespace,

--- a/tests/input3/src/main/scala/example/OrType.scala
+++ b/tests/input3/src/main/scala/example/OrType.scala
@@ -1,0 +1,3 @@
+package example
+
+type X = String | Int

--- a/tests/unit/src/test/resources/documentSymbol-scala3/example/OrType.scala
+++ b/tests/unit/src/test/resources/documentSymbol-scala3/example/OrType.scala
@@ -1,0 +1,3 @@
+/*example(Package):3*/package example
+
+/*example.X(TypeParameter):3*/type X = String | Int

